### PR TITLE
document property paths

### PR DESCRIPTION
**Pull request by tgummerer**
Write some initial documentation for property paths.

Not super happy with this, but I'm also not quite sure what else to document here.  Suggestions very welcome!

### CODE REVIEW

File: docs/architecture/types/property-path.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Property Paths
+
+[`PropertyValue`s](TODO: document these) in Pulumi are often grouped together either as a map of `PropertyKey` to `PropertyValue`, or as a single object property value.
```
Comment (Frassle) [inline docs/architecture/types/property-path.md:3]: Worth pointing out this is a `PropertyMap`

File: docs/architecture/types/property-path.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+root.nested.array[0].double[1]
+root["key with \"escaped\" quotes"]
+root["key with a ."]
+["root key with \"escaped\" quotes"].nested
+["root key with a ."][100]
+root.array[*].field
+root.array["*"].field
+```
+
+Conveniently we can turn a `map[PropertyKey]PropertyValue` into a `PropertyValue` to be addressed by a `PropertyPath`, by just using `NewObjectProperty`.
```
Comment (Frassle) [inline docs/architecture/types/property-path.md:27]:
```suggestion
Conveniently we can turn a `map[PropertyKey]PropertyValue` (aka `PropertyMap`) into a `PropertyValue` to be addressed by a `PropertyPath`, by just using `NewObjectProperty`.
```

File: docs/architecture/types/property-path.md (hunk 1 above)
Comment (tgummerer) [reply docs/architecture/types/property-path.md]: Done :heavy_check_mark:

File: docs/architecture/types/property-path.md (hunk 2 above)
Comment (tgummerer) [reply docs/architecture/types/property-path.md]: Thanks :heavy_check_mark:

**Review by Frassle — APPROVED** [review]

**Comment by Frassle** [comment]
Good start :) we can add more to this as questions come up

**Pull request merged**